util/include/Metrics.hpp

```diff
@@ -107,28 +107,13 @@ class Counter {
 };
 
 class Summary {
-  std::shared_ptr<prometheus::Summary> summary_;
+  ISummary* summaryImp_;
 
  public:
-  struct Quantile {
-    double quantile = 0.0;
-    double value = 0.0;
-    Quantile(double quantile_, double value_) : quantile(quantile_), value(value_) {}
-  };
-
-  struct SummaryDescription {
-    std::uint64_t sample_count = 0;
-    double sample_sum = 0.0;
-    std::vector<Quantile> quantile;
-  };
-
-  using InitQuantiles = std::vector<std::tuple<double, double>>;
-
-  Summary(const InitQuantiles& quantiles);
-  void Observe(double value);
-  SummaryDescription Collect();
-
-  std::string ToJson();
+  Summary(ISummary* summary) : summaryImp_(summary) {}
```

Comment:
Please use `unique_ptr<ISummary>` and get rid of the destructor.

Reply:
Concord bft metrics framework makes some copy by value, return by value, and reference operations.
In this case, using uniqe_ptr may cause problems (for example, how to define a summary handler, as a reference to unique_ptr?).
This why I prefer to keep using raw pointers in this case...
What do you think?

Reply:
How does a raw pointer help you with this problem vs a unique_ptr? Maybe I'm not following the problem, but if you want a reference or value to be returned from any pointer, including smart pointers, can't you just dereference it with `*`?